### ISSUE TITLE
can't f5 into latest master

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#38968]
When i try, i get this error in the VS that has launched:

```

SetSite failed for package [CSharpPackage]Source: 'Microsoft.VisualStudio.LanguageServices' Description: Could not load file or assembly 'System.Collections.Immutable, Version=1.2.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040) System.IO.FileLoadException: Could not load file or assembly 'System.Collections.Immutable, Version=1.2.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040) File name: 'System.Collections.Immutable, Version=1.2.4.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'   at Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService.AbstractPackage`2.<InitializeAsync>d__8.MoveNext()   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[TStateMachine](TStateMachine& stateMachine)   at Microsoft.VisualStudio.LanguageServices.Implementation.LanguageService.AbstractPackage`2.InitializeAsync(CancellationToken cancellationToken, IProgress`1 progress)   at Microsoft.VisualStudio.LanguageServices.CSharp.LanguageService.CSharpPackage.<>n__0(CancellationToken cancellationToken, IProgress`1 progress)   at Microsoft.VisualStudio.LanguageServices.CSharp.LanguageService.CSharpPackage.<InitializeAsync>d__2.MoveNext() --- End of stack trace from previous location where exception was thrown ---    at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)   at Microsoft.VisualStudio.Shell.AsyncPackage.<>c__DisplayClass19_0.<<Microsoft-VisualStudio-Shell-Interop-IAsyncLoadablePackageInitialize-Initialize>b__1>d.MoveNext() --- End of stack trace from previous location where exception was thrown ---   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)   at Microsoft.VisualStudio.Threading.JoinableTask.<JoinAsync>d__68.MoveNext() --- End of stack trace from previous location where exception was thrown ---    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()   at Microsoft.VisualStudio.Services.VsTask.RethrowException(AggregateException e)   at Microsoft.VisualStudio.Services.VsTask.InternalGetResult(Boolean ignoreUIThreadCheck)   at Microsoft.VisualStudio.Services.VsTask.GetResult() WRN: Assembly binding logging is turned OFF. To enable assembly bind failure logging, set the registry value [HKLM\Software\Microsoft\Fusion!EnableLog] (DWORD) to 1. Note: There is some performance penalty associated with assembly bind failure logging. To turn this feature off, remove the registry value [HKLM\Software\Microsoft\Fusion!EnableLog].
--
```

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-infrastructure @jasonmalinowski .

**Comment by jinujoseph** [comment]
cc @RikkiGibson

**Comment by RikkiGibson** [comment]
@agocke is this possibly related to #38809?

**Comment by RikkiGibson** [comment]
I don't know a lot about how we load our dependencies when we run in VS. But I'm wondering if this has the same root cause as to why we can't insert the latest master into VS.

**Comment by RikkiGibson** [comment]
I believe this issue is fixed for now since #38809 was reverted. I just tested f5-ing roslyn master using VS 16.3.2 at commit 7bf4366 and it seems to work.

I think this is a scenario we need to test for whoever does #39017

**Comment by RikkiGibson** [comment]
@CyrusNajmabadi I am wondering how we actually broke f5, because it doesn't match my expectation.

I just tried checking out commit c350cf34c7 (which uses 1.2.4.0 of System.Collections.Immutable), then opened Roslyn.sln using Visual Studio 16.3.2 (latest stable public release). But f5 seems to work fine, can load a simple project and use the editor.

It might be useful to know which commit you had checked out when you experienced this issue. Or if there's anything else significant that differs about your environment that would make it work in my setup but not yours.
